presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1055,6 +1056,11 @@ private static boolean deleteIfExists(FileSystem fileSystem, Path path)
      */
     private List<String> recursiveDeleteFilesStartingWith(String location, String filePrefix)
     {
+        return recursiveDeleteFilesStartingWith(location, filePrefix, new String[0]);
+    }
+
+    private List<String> recursiveDeleteFilesStartingWith(String location, String filePrefix, String[] excludePrefixes)
```

Comment:
Lets always prevent deleting files that start with the hidden file prefixes, and then just verify the filePrefix does not start with a hidden prefix.